app/models/post.rb

```diff
@@ -0,0 +1,34 @@
+class Post < ApplicationRecord
+  belongs_to :submitter, class_name: :User, foreign_key: :user_id
+
+  has_many :comments
+  has_many :direct_comments, -> { where(comment_id: nil) }, class_name: "Comment"
```

Comment:
Did you consider using [polymorphic associations](http://guides.rubyonrails.org/association_basics.html#polymorphic-associations) here?

Reply:
See thread below regarding new comment form partial [https://github.com/gingin77/hackernews_clone/pull/15#pullrequestreview-95941585](url) 

https://github.com/gingin77/hackernews_clone/issues/18